src/security/intel/cbnt/measurement.c

```diff
@@ -0,0 +1,675 @@
+ *
+ * Versions in structures could be different for other revisions of the document.
+ */
+
+/* Header of the Boot Policy Manifest. */
+struct bpm_header {
+	char structure_id[8];     /* "__ACBP__" */
+	uint8_t struct_ver;       /* 0x24 */
+	uint8_t hdr_struct_ver;   /* 0x20 */
+	uint16_t hdr_size;
+	uint16_t key_sig_offset;  /* offset to KeySignature field of Policy Manifest Signature
+				     Element (PMSE) */
+	uint8_t bpm_revision;
+	uint8_t bpm_svn;
+	uint8_t acmsvn_auth;
+	uint8_t reserved;
+	uint16_t nem_data_stack;  /* in 4 KiB pages */
+	uint8_t se_element[];
+} __packed;
+
+/* Deprecated uses must be initialized as { .alg = TPM_ALG_NULL (0x10), .size = 0 } */
+struct hash_struct {
+	uint16_t alg;
+	uint16_t size;
+	uint8_t data[];
+} __packed;
+
+struct bpm_hash_list {
+	uint16_t size;
+	uint16_t count;
+	struct hash_struct hashes[];
+} __packed;
+
+/* IBB Segment Element (upper part) */
+struct bpm_ibbs {
+	char structure_id[8];  /* "__IBBS__" */
+	uint8_t struct_ver;    /* 0x21 */
+	uint8_t reserved1;
+	uint16_t element_size;
+	uint8_t reserved2;
+	uint8_t set_number;    /* 0 */
+	uint8_t reserved3;
+	uint8_t pbet_value;
+	uint32_t flags;        /* see BPM_IBBS_FLAG_* constants for values */
+	uint64_t iommu_bar0;
+	uint64_t iommu_bar1;
+	uint32_t dma_prot_base0;
+	uint32_t dma_prot_limit0;
+	uint64_t dma_prot_base1;
+	uint64_t dma_prot_limit1;
+	struct hash_struct post_ibb_hash; /* deprecated since v1.2.0 of CBnT BWG */
+	uint32_t ibb_entry_point;
+	struct bpm_hash_list digest_list; /* each entries is of variable size */
+	/* struct bpm_ibbs_bottom bottom; */
+} __packed;
+
+/* IBB Segment Element (lower part) */
+struct bpm_ibbs_bottom {
+	struct hash_struct obb_hash; /* deprecated since v1.2.0 of CBnT BWG */
+	uint8_t reserved4[3];
+	uint8_t segment_count;
+	/* ibb_segments[segment_count]; */
+} __packed;
+
+/* KMHASH_STRUCT */
+struct km_hash {
+	uint64_t usage;           /* see KM_HASH_USAGE_* constants for values */
+	struct hash_struct hash;
+} __packed;
+
+/* Key Manifest */
+struct km_header {
+	char structure_id[8];       /* "__KEYM__" */
+	uint8_t struct_ver;         /* 0x21 */
+	uint8_t reserved1[3];
+	uint16_t key_sig_offset;
+	uint8_t reserved2[3];
+	uint8_t km_revision;
+	uint8_t km_svn;
+	uint8_t km_id;
+	uint16_t km_pub_key_hash_alg;
+	uint16_t key_count;
+	struct km_hash key_hash[];  /* of variable size */
+	/* key_sig_offset points here where Key Manifest Signature data is one
+	   of 2 Key Signature Structure formats: RSA or ECC / SM2. */
+} __packed;
+
+static const void *find_in_fit(uint8_t type, size_t *size)
+{
+	struct fit_entry *entry = fit_table_search(type);
+	if (entry == NULL) {
+		printk(BIOS_ERR, "CBnT: failed to find FIT entry: %#02x\n", type);
+		return NULL;
+	}
+
+	void *addr = (void *)(uintptr_t)entry->address.u64;
+
+	switch (type) {
+		/* All known versions of the header (up to v5.4) have compatible version and
+		   size fields. */
+		const struct acm_header_v0 *acm;
```

Comment:
This should be on top of the function.

Reply:
Moved the declaration before the `switch`: https://github.com/Dasharo/coreboot/compare/84fb089a040e938ced09b2d74d1b5dd9d17e3348..4a20083b3990dba0883f92b1f90408bbbb276a61#diff-1725d4cd7234cc446584ac6ebad791ed8c8e640970d4adb2e99e5b5b233030f7

Top of the function would be farther away from the use and inconsistent with other declarations in this and other functions.